tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/query-store-dialog.js

```diff
@@ -77,6 +79,35 @@ define(['require', 'lodash', 'jquery', 'log', 'backbone', '../../../js/event-sim
                         }
                     );
 
+                    appNameSelector.on('change', function (event) {
+                        var siddhiAppName = storeQueryModal.find("select[name='siddhi-app-name']").val();
+                        if (siddhiAppName != 'undefined' && sessionStorage.getItem("onDemandTempStore") != null) {
+                            var onDemandTempStore = JSON.parse(sessionStorage.getItem("onDemandTempStore"));
+                            if (onDemandTempStore[siddhiAppName] != null) {
+                                queryTextArea.val(onDemandTempStore[siddhiAppName]);
+                            } else {
+                                queryTextArea.val('');
+                            }
+                        }
+                    });
+
+                    queryTextArea.on('input change keyup', function (event) {
```

Comment:
Let's store the query only if the execute button is pressed rather than storing for each key press